packages/typespec-ts/src/transform/transformResponses.ts

```diff
@@ -88,12 +98,21 @@ export function transformToResponseTypes(
       // transform header
       const headers = transformHeaders(dpgContext, resp, inputImportedSet);
       // transform body
-      const body = transformBody(dpgContext, resp, inputImportedSet);
+      const [body, schemas] = transformBody(
+        dpgContext,
+        resp,
+        inputImportedSet
+      ) ?? [undefined, []];
       rlcOperationUnit.responses.push({
         ...rlcResponseUnit,
         headers,
         body
       });
+      if (statusCode === "default") {
+        defaultSchemas.push(...schemas);
+      } else {
+        nonDefaultSchemas.push(...schemas);
```

Comment:
as we know currently we only consider the case for sub model cases, actually it may happen for other sub types like string, union ect. using schema could not change the contract here if we plan to consider other cases.